### PR TITLE
Replace FlatBuffers fork with upstream 2.0.5

### PR DESCRIPTION
**Pull request by vaind**
TODO:
* [x] verify performance in the flutter benchmark (first try seems to show a slight degradation on readAll

### CODE REVIEW

**Comment by vaind** [comment]
# Rough benchmark results
confirms that readAll is slightly slower after this PR.

## main @ 6d3bc38370348ee45e035d7ce41240589249559e
```yaml
dependency_overrides:
  objectbox:
    git:
      url: git@github.com:objectbox/objectbox-dart.git
      path: objectbox
  objectbox_flutter_libs:
    git:
      url: git@github.com:objectbox/objectbox-dart.git
      path: flutter_libs
  objectbox_generator:
    git:
      url: git@github.com:objectbox/objectbox-dart.git
      path: generator
```
image image


## this branch (FB 2.0.5)
image image

**Comment by vaind** [comment]
After a fix to pass `asciOptimization: true` to `fb.StringReader()` (was true by default in the fork):

image

**Comment by vaind** [comment]
Good to be merged from my POV, the outstanding failures in CI (flutter beta) are unrelated to this PR

File: generator/lib/src/code_chunks.dart [hunk 1]
```diff
@@ -331,24 +335,25 @@ class CodeChunks {
...
           } else {
             fbReader = 'fb.ListReader<int>(fb.Int8Reader(), lazy: false)';
           }
           break;
         case OBXPropertyType.Relation:
           fbReader = 'fb.${_propertyFlatBuffersType[p.type]}Reader()';
           return readFieldNonNull('0');
+        case OBXPropertyType.String:
+          // TODO `asciiOptimization: true` is for keeping the same behavior as the previous FB fork.
+          //      Check if it still makes sense with the latest Dart/Flutter.
```
Comment (greenrobot-team) [inline generator/lib/src/code_chunks.dart:349]: Were these TODOs resolved? Or still needs to be done, e.g. create an issue?

**Review by greenrobot-team — COMMENTED** [review]
LGTM, just a question about added TODOs.

File: generator/lib/src/code_chunks.dart (hunk 1 above)
Comment (vaind) [reply generator/lib/src/code_chunks.dart:349]: Created an issue #360

**Pull request merged**